### PR TITLE
allow for unsorted integer array indices, and integer indices with duplicates

### PR DESCRIPTION
**Pull request by jswhit**
addresses issue #467 by standardizing slicing code around integer arrays instead of boolean arrays.

### CODE REVIEW

File: netCDF4/utils.py [hunk 1]
```diff
@@ -267,22 +258,19 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
...
-                if len(indices) >= 2:
-                    step = indices[1] - indices[0]
-                else:
-                    step = None
-                if np.array_equal(indices, np.arange(start, stop, step)):
-                    newElem.append(slice(start, stop, step))
-                else:
-                    newElem.append(ea)
+        # Replace sequence of indices with slice object if possible.
+        elif np.iterable(e) and len(e) > 1:
```
Comment (shoyer) [inline netCDF4/utils.py]: I would consider adding another case for `len(e) == 1`, which I expect is actually somewhat common. We could replace that with `slice(e[0], e[0] + 1, 1)`.

File: netCDF4/utils.py [hunk 2]
```diff
@@ -267,22 +258,19 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
...
+            start = e[0]
+            stop = e[-1]+1
+            step = e[1]-e[0]
+            try:
+                ee = range(start,stop,step)
```
Comment (shoyer) [inline netCDF4/utils.py]: I would add a `range = xrange` at the top of this file on Python 2.  That means you don't actually need to allocate a Python list to check the length of the range.

**Comment by shoyer** [comment]
look very nice, thanks!

File: netCDF4/utils.py (hunk 2 above)
Comment (jswhit) [inline netCDF4/utils.py]: Done.

File: netCDF4/utils.py (hunk 1 above)
Comment (jswhit) [inline netCDF4/utils.py]: Done.

**Comment by jhamman** [comment]
Thanks @jswhit!

**Pull request merged**